### PR TITLE
Fix the census initialization on the server. Remove duplicate initialization logic and also enable census per channel if census is initialized.

### PR DESCRIPTION
**Pull request by bogdandrutu**
With this change we can enable census on the server by initializing the census library. At one point we should probably send the client channel arguments and use them for each channel to control the census library.

### CODE REVIEW

File: src/core/surface/channel_create.c [hunk 1]
```diff
@@ -200,7 +201,7 @@ grpc_channel *grpc_insecure_channel_create(const char *target,
       "grpc_insecure_channel_create(target=%p, args=%p, reserved=%p)", 3,
       (target, args, reserved));
   GPR_ASSERT(!reserved);
-  if (grpc_channel_args_is_census_enabled(args)) {
+  if (grpc_channel_args_is_census_enabled(args) || census_enabled()) {
```
Comment (a-veitch) [inline src/core/surface/channel_create.c]: Shouldn't this be && ? Or can the census_enabled() check be dropped, and made part of whatever grpc_channel_args_is_census_enabled checks?

File: src/core/surface/secure_channel_create.c [hunk 2]
```diff
@@ -295,7 +296,7 @@ grpc_channel *grpc_secure_channel_create(grpc_channel_credentials *creds,
   args_copy = grpc_channel_args_copy_and_add(
       new_args_from_connector != NULL ? new_args_from_connector : args,
       &connector_arg, 1);
-  if (grpc_channel_args_is_census_enabled(args)) {
+  if (grpc_channel_args_is_census_enabled(args) || census_enabled()) {
```
Comment (a-veitch) [inline src/core/surface/secure_channel_create.c]: ditto.

File: src/core/surface/server.c [hunk 3]
```diff
@@ -779,9 +780,8 @@ grpc_server *grpc_server_create_from_filters(
     const grpc_channel_filter **filters, size_t filter_count,
     const grpc_channel_args *args) {
   size_t i;
-  /* TODO(census): restore this once we finalize census filter etc.
-     int census_enabled = grpc_channel_args_is_census_enabled(args); */
-  int census_enabled = 0;
+  int enable_census =
```
Comment (a-veitch) [inline src/core/surface/server.c]: make this a bool?

Comment (a-veitch) [inline src/core/surface/server.c]: better name might be census_is_enabled.

File: src/core/surface/server.c [hunk 4]
```diff
@@ -779,9 +780,8 @@ grpc_server *grpc_server_create_from_filters(
...
+      grpc_channel_args_is_census_enabled(args) || census_enabled();
```
Comment (a-veitch) [inline src/core/surface/server.c]: ditto on the || -> &&

Comment (bogdandrutu) [inline src/core/surface/server.c]: Fixed.

File: src/core/surface/secure_channel_create.c (hunk 2 above)
Comment (bogdandrutu) [inline src/core/surface/secure_channel_create.c]: Fixed.

File: src/core/surface/channel_create.c (hunk 1 above)
Comment (bogdandrutu) [inline src/core/surface/channel_create.c]: Fixed.

File: src/core/surface/server.c (hunk 3 above)
Comment (bogdandrutu) [inline src/core/surface/server.c]: kept the initial name no conflict anymore.

File: src/core/surface/server_create.c [hunk 5]
```diff
@@ -43,9 +43,6 @@ grpc_server *grpc_server_create(const grpc_channel_args *args, void *reserved) {
   const grpc_channel_filter *filters[3];
   size_t num_filters = 0;
   filters[num_filters++] = &grpc_compress_filter;
-  if (grpc_channel_args_is_census_enabled(args)) {
-    filters[num_filters++] = &grpc_server_census_filter;
-  }
```
Comment (a-veitch) [inline src/core/surface/server_create.c]: Why is this removed? Where does the census filter get installed if census is enabled?

Comment (bogdandrutu) [inline src/core/surface/server_create.c]: In the grpc_server_create_from_filters which is in the server.c . You commented that line then ctiller added this and we end up with 2 initialization logic but one was never gonna work.
